src/utils/RNPxlSearch.cpp

```diff
@@ -2350,13 +2162,684 @@ class RNPxlSearch :
+    for (int z = 1; z <= precursor_charge; ++z)
+    {
+      // 1. create unshifted peaks (a,b,y, MS2 precursor ions up to pc charge)
+      PeakSpectrum tmp_shifted_series_peaks;
+      partial_loss_spectrum_generator.getSpectrum(tmp_shifted_series_peaks, fixed_and_variable_modified_peptide, z, z);
+
+      PeakSpectrum::StringDataArray& tmp_shifted_series_annotations = tmp_shifted_series_peaks.getStringDataArrays()[0];
+      PeakSpectrum::IntegerDataArray& tmp_shifted_series_charges = tmp_shifted_series_peaks.getIntegerDataArrays()[0];
+
+      // 2. shift peaks 
+      for (auto & p : tmp_shifted_series_peaks) { p.setMZ(p.getMZ() + fragment_shift_mass / static_cast<double>(z)); }
+
+      // 3. add shifted peaks to shifted_series_peaks
+      shifted_series_peaks.insert(shifted_series_peaks.end(), tmp_shifted_series_peaks.begin(), tmp_shifted_series_peaks.end());
+      shifted_series_annotations.insert(
+        shifted_series_annotations.end(),
+        tmp_shifted_series_annotations.begin(),
+        tmp_shifted_series_annotations.end()
+      );
+      shifted_series_charges.insert(
+        shifted_series_charges.end(),
+        tmp_shifted_series_charges.begin(),
+        tmp_shifted_series_charges.end()
+      );
+    }
+
+    // 4. add fragment shift name to annotation of shifted peaks
+    for (Size j = 0; j != shifted_series_annotations.size(); ++j)
+    {
+      shifted_series_annotations[j] = shifted_series_annotations[j] + " " + fragment_shift_name;
+    }
+
+    // append shifted and annotated ion series to partial loss spectrum
+    partial_loss_spectrum.insert(partial_loss_spectrum.end(), shifted_series_peaks.begin(), shifted_series_peaks.end());
+    partial_loss_spectrum_annotation.insert(
+      partial_loss_spectrum_annotation.end(),
+      shifted_series_annotations.begin(),
+      shifted_series_annotations.end()
+    );
+    partial_loss_spectrum.getIntegerDataArrays()[0].insert(
+      partial_loss_spectrum_charge.end(),
+      shifted_series_charges.begin(),
+      shifted_series_charges.end()
+    );
+  }
+
+  partial_loss_spectrum.sortByPosition();
+}
+
+void RNPxlSearch::RNPxlFragmentIonGenerator::addPrecursorWithCompleteRNA_(
+  const double fixed_and_variable_modified_peptide_weight, 
+  const String &precursor_rna_adduct,
+  const double precursor_rna_weight, 
+  const int charge, 
+  PeakSpectrum &partial_loss_spectrum,
+  MSSpectrum::IntegerDataArray &partial_loss_spectrum_charge,
+  MSSpectrum::StringDataArray &partial_loss_spectrum_annotation)
+{
+  const double xl_mz = (fixed_and_variable_modified_peptide_weight + precursor_rna_weight +
+                  static_cast<double>(charge) * Constants::PROTON_MASS_U)
+                 / static_cast<double>(charge);
+  partial_loss_spectrum.push_back(Peak1D(xl_mz, 1.0));
+  partial_loss_spectrum_charge.push_back(charge);
+  partial_loss_spectrum_annotation.push_back(String("[M+") + precursor_rna_adduct + "]");
+}
+
 int main(int argc, const char** argv)

```

Comment:
this class / file looks like its way to large. Is there a way that some this can make its way into the library?